activerecord/test/cases/schema_dumper_test.rb

```diff
@@ -497,6 +497,86 @@ def change
 
     $stdout = original
   end
+
+  if current_adapter?(:PostgreSQLAdapter)
+    def test_schema_dump_with_correct_timestamp_types_via_create_table_and_t_column
+      original, $stdout = $stdout, StringIO.new
+
+      migration = Class.new(ActiveRecord::Migration::Current) do
+        def up
+          create_table("timestamps") do |t|
+            t.datetime :default_format
+            t.column :without_time_zone, :timestamp
+            t.column :with_time_zone, :timestamptz
+          end
+        end
+        def down
+          drop_table("timestamps")
+        end
+      end
+      migration.migrate(:up)
+
+      output = perform_schema_dump
+      assert output.include?('t.datetime "default_format"')
+      assert output.include?('t.datetime "without_time_zone"')
+      assert output.include?('t.timestamptz "with_time_zone"')
```

Comment:
without the changes in this PR, this test would fail. `output.include?('t.datetime "with_time_zone"')` would be true. the problem then is when you recreate the table (eg. in a test environment), the column is created as `timestamp`, not `timestamptz`.